### ISSUE TITLE
UIApplication.statusBarOrientation = Unknown

### ISSUE THREAD

**Issue by minixxie** [skywinder/ActionSheetPicker-3.0#152]
Hi Skywinder,

I'm trying the DatePicker on my iphone4 (build with iOS 8.1 SDK while running on iOS 7.1.2(11D257), the date picker is too wide (the Done button went out of the screen too), screenshot:
![img_0001](https://cloud.githubusercontent.com/assets/1013500/6754047/164d0b86-cf55-11e4-8401-aa1bdf9bf82e.PNG)

After investigation, I found that AbstractActionSheetPicker::isViewPortrait function is returning false.
The reason is:
value of [UIApplication sharedApplication].statusBarOrientation = 0 (Unknown)
My phone was placing flat on the desk. When I pick it up straight, it becomes correct returning UIInterfaceOrientationPortrait.

Also, if I add fix into this function by:
if ( [UIApplication sharedApplication].statusBarOrientation == UIInterfaceOrientationUnknown )
    {
        NSLog(@"isViewPortrait: fixing to portrait...");
        return YES;
    }
The width of the UIDatePicker is still too wide and went out of the screen. But I can see that that "Done" button was fixed with this, which was out of the screen without the above fix.

Am I missing anything? Thank you.

Regards,
Simon

**Comment by skywinder** [comment]
Hello! Thanks for mention it! I Will try investigate it later.
Anyway, if you find solution - please, feel free to create a PR with fix!

**Comment by minixxie** [comment]
Sorry, I'm really not sure if this is a fix. And I just confirmed that altho the above if-statement was added, it only fixed the "Done" button (thus the width of the white color part), but not the UIDatePicker part.

And I was calling this to fixed both the "Done" button and the UIDatePicker's width: (calling before creating the ActionSheetDatePicker)
	[UIApplication sharedApplication].statusBarOrientation = UIInterfaceOrientationPortrait;//hacky fix

I'm not sure if this is a correct fix. As this should mean that my application is ALWAYS portrait. That shouldn't be true for application that intended to auto rotate.

Regards,
Simon
